src/measure_theory/integral/average.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2022 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury G. Kudryashov
+-/
+import measure_theory.integral.set_integral
+
+/-!
+# Integral average of a function
+
+In this file we define `measure_theory.average μ f` (notation: `⨍ x, f x ∂μ`) to be the average
+value of `f` with respect to measure `μ`. It is defined as `∫ x, f x ∂((μ univ)⁻¹ • μ)`, so it
+is equal to zero if `f` is not integrable or if `μ` is an infinite measure. If `μ` is a probability
+measure, then the average of any function is equal to its integral.
+
+For the average on a set, we use `⨍ x in s, f x ∂μ` (notation for `⨍ x, f x ∂(μ.restrict s)`). For
+average w.r.t. the volume, one can omit `∂volume`.
+
+## Implementation notes
+
+The average is defined as an integral over `(μ univ)⁻¹ • μ` so that all theorems about Bochner
+integrals work for the average without modifications. For theorems that require integrability of a
+function, we provide a convenience lemma `measure_theory.integrable.to_average`.
+
+## Tags
+
+integral, center mass, average value
+-/
+
+open measure_theory measure_theory.measure metric set filter topological_space function
+open_locale topological_space big_operators ennreal convex
+
+variables {α E F : Type*} {m0 : measurable_space α}
+  [normed_group E] [normed_space ℝ E] [complete_space E]
+  [second_countable_topology E] [measurable_space E] [borel_space E]
+  [normed_group F] [normed_space ℝ F] [complete_space F]
+  [second_countable_topology F] [measurable_space F] [borel_space F]
+  {μ : measure α} {s : set E}
+
+/-!
+### Average value of a function w.r.t. a measure
+
+The average value of a function `f` w.r.t. a measure `μ` (notation: `⨍ x, f x ∂μ`) is defined as
+`(μ univ).to_real⁻¹ • ∫ x, f x ∂μ`, so it is equal to zero if `f` is not integrable or if `μ` is an
+infinite measure. If `μ` is a probability measure, then the average of any function is equal to its
+integral.
+
+-/
+
+namespace measure_theory
+
+variable (μ)
+include m0
+
+/-- Average value of a function `f` w.r.t. a measure `μ`, notation: `⨍ x, f x ∂μ`. It is defined as
+`(μ univ).to_real⁻¹ • ∫ x, f x ∂μ`, so it is equal to zero if `f` is not integrable or if `μ` is an
+infinite measure. If `μ` is a probability measure, then the average of any function is equal to its
+integral.
+
+For the average on a set, use `⨍ x in s, f x ∂μ` (defined as `⨍ x, f x ∂(μ.restrict s)`). For
+average w.r.t. the volume, one can omit `∂volume`. -/
+noncomputable def average (f : α → E) := ∫ x, f x ∂((μ univ)⁻¹ • μ)
+
+notation `⨍` binders `, ` r:(scoped:60 f, f) ` ∂` μ:70 := average μ r
+notation `⨍` binders `, ` r:(scoped:60 f, average volume f) := r
```

Comment:
What's the difference between this and the following?
```suggestion
notation `⨍` binders `, ` r:(scoped:60 f, f) := average volume r
```
(not actually asking you to change this, just curious)